### PR TITLE
Config management for credentials post login operation

### PR DESCRIPTION
**Pull request by rsharmila10vmware**

### CODE REVIEW

File: background/main.go [hunk 1]
```diff
@@ -87,6 +113,7 @@ func placeHolderApi() {
 func DeployToVcenter() {
 	log.Println("Starting deployment to vcenter with sample details ")
 	_ = Marketplace.DeployVcenter()
+	time.Sleep(60 * time.Second)
```
Comment (arahamansm) [inline background/main.go]: why we have introduced sleep here ?

File: internal/models/agent.go [hunk 2]
```diff
@@ -9,9 +9,9 @@ type VCenterRegistration struct {
 }
 
 type AgentLoginRequest struct {
-	UserID         string `json:"userid"`
+	UserID         string `json:"userId"`
```
Comment (arahamansm) [inline internal/models/agent.go:12]: why have we changed the casing here, i think our rest-api request expects in small case ?

File: pkg/agent.go [hunk 3]
```diff
@@ -2,56 +2,96 @@ package pkg
...
-		return agentLoginRes, fmt.Errorf("failed to perform agent login %s : (%d)", agentLoginReq, resp.StatusCode)
+		log.Println("Agent not registered with Marketplace. Generating new reference ID for agent")
+		m.AgentConfig.CLIID = uuid.String()
+		m.AgentConfig.Flush() // push the registered UUID to config for further interactions with Marketplace
 	}
 
-	err = m.DecodeJson(resp.Body, agentLoginRes)
+	// call marketplace API for login
+	response, err := m.AgentOperations.AgentLogin(&models.AgentLoginRequest{
+		UserID:         "rsharmila",
```
Comment (arahamansm) [inline pkg/agent.go:33]: from where we will get this userId , have we thought about it ?

File: internal/models/agent.go (hunk 2 above)
Comment (rsharmila10vmware) [reply internal/models/agent.go:12]: this is the confirmation i was asking today. When I test, its expecting camel cased.

File: pkg/agent.go (hunk 3 above)
Comment (rsharmila10vmware) [reply pkg/agent.go:33]: as I told earlier, the oAuth app doesn't have a user id, so until we get some native identifiers from on prem, we had decided to hardcode

Comment (arahamansm) [reply pkg/agent.go:33]: ok

File: internal/models/agent.go (hunk 2 above)
Comment (arahamansm) [reply internal/models/agent.go:12]: let me check

**Review by arahamansm — APPROVED** [review]

**Pull request merged**